src/ec2/prompter.ts

```diff
@@ -4,43 +4,62 @@
  */
 
 import { RegionSubmenu, RegionSubmenuResponse } from '../shared/ui/common/regionSubmenu'
-import { Ec2Selection, getInstancesFromRegion } from './utils'
+import { Ec2Selection } from './utils'
 import { DataQuickPickItem } from '../shared/ui/pickerPrompter'
-import { Ec2Instance } from '../shared/clients/ec2Client'
+import { Ec2Client, Ec2Instance } from '../shared/clients/ec2Client'
 import { isValidResponse } from '../shared/wizards/wizard'
 import { CancellationError } from '../shared/utilities/timeoutUtils'
+import { AsyncCollection } from '../shared/utilities/asyncCollection'
 
-function asQuickpickItem(instance: Ec2Instance): DataQuickPickItem<string> {
-    return {
-        label: '$(terminal) \t' + (instance.name ?? '(no name)'),
-        detail: instance.InstanceId,
-        data: instance.InstanceId,
+export type instanceFilter = (instance: Ec2Instance) => boolean
+
+export class Ec2Prompter {
```

Comment:
should this `extends Prompter` like most other FooPrompter classes?

    git grep -P 'class.*Prompter'

Reply:
This case might be unique. Since we're leveraging the `regionSubmenu`, we have already extended the prompter class there. This class serves as a way to organize the methods relating to the EC2 implementation of `regionSubmenu`.